clients/context-registry/src/test/integration/ContextRegistryClient.test.ts

```diff
@@ -4,68 +4,121 @@
-  const contextRegistry: ContextRegistryClient = new ContextRegistryClient();
+  const iTwinAccessClient: ITwinAccessClient = new ITwinAccessClient();
   let requestContext: AuthorizedClientRequestContext;
 
   before(async function () {
     this.timeout(0);
     requestContext = await TestConfig.getAuthorizedClientRequestContext();
   });
 
-  it("should get a list of projects (#integration)", async () => {
-    const queryOptions: ContextRegistryRequestQueryOptions = {
-      $select: "*",
-      $top: 20,
-    };
+  it("should get a list of iTwins (#integration)", async () => {
+    const iTwinList: ITwin[] = await iTwinAccessClient.getAll(requestContext);
 
-    const projects: Project[] = await contextRegistry.getProjects(requestContext, queryOptions);
-    chai.expect(projects.length).greaterThan(0);
+    // At least one iTwin
+    chai.expect(iTwinList).to.not.be.empty;
   });
 
-  it("should get a list of Most Recently Used (MRU) projects (#integration)", async () => {
-    const queryOptions: ContextRegistryRequestQueryOptions = {
-      $select: "*",
-      $top: 20,
-      isMRU: true,
-    };
+  it("should get a paged list of iTwins using top (#integration)", async () => {
+    const numberOfITwins = 3;
 
-    const projects: Project[] = await contextRegistry.getProjects(requestContext, queryOptions);
-    chai.expect(projects.length).greaterThan(0);
-  });
+    // Verify there are enough iTwins to test the paging
+    const fullITwinList: ITwin[] = await iTwinAccessClient.getAll(requestContext);
+    chai.assert(fullITwinList.length >= numberOfITwins, "Unable to meaningfully run test since there are too few iTwins.");
 
-  it("should get a list of Favorite projects (#integration)", async () => {
-    const queryOptions: ContextRegistryRequestQueryOptions = {
-      $select: "*",
-      $top: 20,
-      isFavorite: true,
-    };
+    const partialITwinList: ITwin[] = await iTwinAccessClient.getAll(requestContext,
+      {
+        pagination: {
+          top: numberOfITwins,
+        },
+      });
 
-    const projects: Project[] = await contextRegistry.getProjects(requestContext, queryOptions);
-    chai.expect(projects.length).to.be.greaterThan(0);
+    // Get the same number of iTwins as the top param
+    chai.expect(partialITwinList).length(numberOfITwins);
   });
 
-  it("should get a project by name (#integration)", async () => {
-    const queryOptions: ContextRegistryRequestQueryOptions = {
-      $select: "*",
-      $filter: `Name+eq+'${TestConfig.projectName}'`,
-    };
-    const project: Project = await contextRegistry.getProject(requestContext, queryOptions);
-    chai.expect(project.name).equals(TestConfig.projectName);
-  });
+  it("should get a paged list of iTwins using skip (#integration)", async () => {
+    const numberSkipped = 3;
+
+    // Verify there are enough iTwins to test the paging
+    const fullITwinList: ITwin[] = await iTwinAccessClient.getAll(requestContext);

```

Comment:
The `fullITwinList` will be automatically limited to max 2000 (because of the query's behavior, 1000 project + 1000 assets) so if the number of projects/assets reach that limit, this test will fail as even skipping 3 selements will still display a 2000 element full list...

Reply:
I did not know about the 1000 item limit per request. I see what you mean, but I do not know how much of an issue it is since the QA users should not approach this number of iTwins. Currently it is retrieving a total of 13 and I would be wary about the test data if the QA users had several hundreds of iTwins.
Soon the implementation of this class will be imported from the CONNECT team so these tests should be long lived as is.

Reply:
Marking this as resolved since this should not be an issue since QA accounts should not have this many items.